diffkemp/simpll/passes/VarDependencySlicer.cpp

```diff
@@ -46,9 +48,39 @@ PreservedAnalyses VarDependencySlicer::run(Function &Fun,
             continue;
         for (auto &Instr : BB) {
             bool dependent = false;
-            for (auto &Op : Instr.operands()) {
-                if (checkDependency(&Op))
-                    dependent = true;
+            if(auto GEPInstr = dyn_cast<GetElementPtrInst>(&Instr)) {
+                if(dyn_cast<GlobalVariable>(GEPInstr->getOperand(0)) == Variable) {
```

Comment:
This block of code is very similar to the code below. It would make sense to enclose it into a function, something like `isVarIndicesAccess`.